File: a02_new_msg_type/scripts/msg_pub.py
import rospy
from a02_new_msg_type.msg import new_msg

def new_msg_talker():
    pub = rospy.Publisher("new_msg_chatter", new_msg, queue_size=10)
    rospy.init_node("new_msg_talker", anonymous=True)
    
    while not rospy.is_shutdown():
        str = "answer to the universe is "
        pub.publish(new_msg(str, 42))
        rospy.sleep(0.1)

if __name__ == '__main__':
    try:
        new_msg_talker()
    except rospy.ROSInterruptException:
        pass

File: a02_new_msg_type/src/msg_pub.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include "a02_new_msg_type/new_msg.h"
#include "iostream"
#include "sstream"

using namespace std;

int main(int argc, char **argv){
    ros::init(argc, argv, "msg_pub");
    ros::NodeHandle node;
    ros::Publisher pub = node.advertise<a02_new_msg_type::new_msg>("new_msg_chatter",10);
    ros::Rate loop_rate(10);

    while(ros::ok()){
        a02_new_msg_type::new_msg msg;

        std::stringstream ss;
        ss << "ans to the universe is ";
        msg.greeting = ss.str();
        msg.number = 42;

        pub.publish(msg);
        ros::spinOnce();
        loop_rate.sleep();
    }
    return 0;
}